### PR TITLE
Fix CI

### LINKED ISSUE #234: Invalid test group specified in CI
https://github.com/JuliaDiff/SparseDiffTools.jl/blob/f191085e7ab4c5281fb80e4c779ae440b5dcc3cc/.github/workflows/CI.yml#L16

This causes  the CI to pass without running any tests.

### PR DESCRIPTION
**Pull request by gaurav-arya**
Resolves #234. Fixes various typos that crept in when CI was disabled

### CODE REVIEW

**Review by vpuri3 — COMMENTED** [review]
@ChrisRackauckas there's some ArrayInterface/ StaticArrays bug here. can you take a look?

**Comment by vpuri3** [comment]
https://github.com/JuliaDiff/SparseDiffTools.jl/actions/runs/4537212339/jobs/7994820746?pr=235#step:6:389

```julia
  LoadError: MethodError: no method matching Float64(::ForwardDiff.Dual{ForwardDiff.Tag{typeof(Main.##280.staticf), Float64}, Float64, 3})
  Closest candidates are:
    (::Type{T})(::Real, ::RoundingMode) where T<:AbstractFloat at rounding.jl:200
    (::Type{T})(::T) where T<:Number at boot.jl:760
    (::Type{T})(::AbstractChar) where T<:Union{AbstractChar, Number} at char.jl:50
    ...
  Stacktrace:
    [1] convert(#unused#::Type{Float64}, x::ForwardDiff.Dual{ForwardDiff.Tag{typeof(Main.##280.staticf), Float64}, Float64, 3})
      @ Base ./number.jl:7
    [2] macro expansion
      @ ~/.julia/packages/StaticArraysCore/U2Z1K/src/StaticArraysCore.jl:81 [inlined]
    [3] convert_ntuple
      @ ~/.julia/packages/StaticArraysCore/U2Z1K/src/StaticArraysCore.jl:77 [inlined]
    [4] StaticArraysCore.SVector{30, Float64}(x::NTuple{30, ForwardDiff.Dual{ForwardDiff.Tag{typeof(Main.##280.staticf), Float64}, Float64, 3}})
      @ StaticArraysCore ~/.julia/packages/StaticArraysCore/U2Z1K/src/StaticArraysCore.jl:113
    [5] SArray
      @ ~/.julia/packages/StaticArraysCore/U2Z1K/src/StaticArraysCore.jl:117 [inlined]
    [6] StaticArray
      @ ~/.julia/packages/StaticArrays/a4r2v/src/convert.jl:167 [inlined]
    [7] restructure(x::StaticArraysCore.SVector{30, Float64}, y::StaticArraysCore.SVector{30, ForwardDiff.Dual{ForwardDiff.Tag{typeof(Main.##280.staticf), Float64}, Float64, 3}})
      @ ArrayInterface.ArrayInterfaceStaticArraysCoreExt ~/.julia/packages/ArrayInterface/xqHUD/ext/ArrayInterfaceStaticArraysCoreExt.jl:34
    [8] SparseDiffTools.ForwardColorJacCache(f::typeof(Main.##280.staticf), x::StaticArraysCore.SVector{30, Float64}, _chunksize::Nothing; dx::StaticArraysCore.SVector{30, Float64}, tag::Nothing, colorvec::Vector{Int64}, sparsity::SparseArrays.SparseMatrixCSC{Float64, Int64})
      @ SparseDiffTools ~/work/SparseDiffTools.jl/SparseDiffTools.jl/src/differentiation/compute_jacobian_ad.jl:47
 ```

**Comment by gaurav-arya** [comment]
I don't think it's related to ArrayInterface / StaticArrays? There's just a dual number that's trying to be written into a float array cache.

**Comment by gaurav-arya** [comment]
Okay, @vpuri3 you're right that's there some StaticArrays / ArrayInterface shenanigans going on. Various typos also causeed the same error to be thrown at different places in the tests -- those are fixed now.

File: ext/SparseDiffToolsZygote.jl [hunk 1]
```diff
@@ -41,19 +41,19 @@ function SparseDiffTools.numback_hesvec(f, x, v)
 end
 
 function SparseDiffTools.autoback_hesvec!(dy, f, x, v,
-                          cache1 = Dual{typeof(ForwardDiff.Tag(DeivVecTag, eltype(x))),
+                          cache1 = Dual{typeof(ForwardDiff.Tag(DeivVecTag(), eltype(x))),
                                         eltype(x), 1
                                         }.(x,
                                            ForwardDiff.Partials.(Tuple.(reshape(v, size(x))))),
-                          cache2 = Dual{typeof(ForwardDiff.Tag(DeivVecTag, eltype(x))),
+                          cache2 = Dual{typeof(ForwardDiff.Tag(DeivVecTag(), eltype(x))),
```
Comment (gaurav-arya) [inline ext/SparseDiffToolsZygote.jl:48]: There was an inconsistency between using the type here and an instance elsewhere causing an error, so I resolved the inconsistency here -- but should I have gone the other way and just used the data type? @ChrisRackauckas

Comment (vpuri3) [reply ext/SparseDiffToolsZygote.jl:48]: `DeivVecTag()` is good

**Comment by gaurav-arya** [comment]
@ChrisRackauckas I think this can be merged. CI was previously completely down, so this is a strict improvement, and helps unblock #236. The only thing that fails in CI is `test_ad.jl` (which I put at the end of `runtests.jl` so that we can see that everything now passes)  -- that's because of #237.

File: src/differentiation/compute_jacobian_ad.jl [hunk 2]
```diff
@@ -44,7 +44,7 @@ function ForwardColorJacCache(f::F, x, _chunksize = nothing;
         p = adapt.(parameterless_type(x), generate_chunked_partials(x, colorvec, chunksize))
         _t = Dual{T, eltype(x), getsize(chunksize)
                   }.(vec(x), ForwardDiff.Partials.(first(p)))
-        t = ArrayInterface.restructure(x, _t)
+        t = ArrayInterface.restructure(similar(x, eltype(_t)), _t)
```
Comment (gaurav-arya) [inline src/differentiation/compute_jacobian_ad.jl:47]: A hack that gets things to work for now. I don't quite understand the semantics of `restructure`, but if it should automatically convert the type for us then this shouldn't be necessary in theory: see https://github.com/JuliaArrays/ArrayInterface.jl/issues/404. The `similar` can also cause an extra allocation in cases.

**Comment by ChrisRackauckas** [comment]
Bump the ArrayInterface lower bound to include the patch.

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**